add functions to work with date and time

uladhsi opened the pull request:
This PR implements the following reuse APIs:

- `util.formatter.formatDateWithTime`
- `common.date.calculateWithTime`
- `common.date.getCurrentDateAndTime`
- `ui5.date.pickWithTime`

DIFSRIP commented on src/reuse/modules/ui5/date.ts:
```diff
@@ -3,6 +3,14 @@
+type PickerMetadata = "sap.m.DatePicker" | "sap.m.DateTimePicker" | "sap.m.DateRangeSelection";
+
+class DatePickerSelectorParams {
+  selector: any;
```
You can use the correct type: 
https://github.com/SAP/wdio-qmate-service/blob/main/src/reuse/modules/ui5/types/ui5.types.ts#L5

```suggestion
  selector: Ui5Selector;
```

DIFSRIP replied:
same everywhere

uladhsi replied:
done: https://github.com/SAP/wdio-qmate-service/pull/418/commits/a4de73df333b8f64dfb410cd3387923cbfb6a2b4

DIFSRIP commented on src/reuse/modules/ui5/date.ts:
```diff
@@ -24,18 +32,7 @@ export class DateModule {
-      }
-    };
-
+    const tempSelector = await this._constructDatePickerSelector({ selector, index, metadata: "sap.m.DatePicker" });
```
not sure why we used the name `tempSelector`? Should we change that to something more meaningful?

uladhsi replied:
renamed to `datePickerSelector` in all the methods of that class: https://github.com/SAP/wdio-qmate-service/pull/418/commits/321003e5da6c3183463ec17a02b32b455c7cc9c4

DIFSRIP reviewed (changes requested):
Great job! Code looks well structured and very flexible implementation :)

I have just added a few minor comments.

I also have another question/suggestion: Would it make sense to support some string value "currentTime" similar as we do for the day ("today", "tomorrow", etc.) for the time as well?

uladhsi commented on src/reuse/modules/common/date.ts:
```diff
@@ -178,5 +189,83 @@ export class DateModule {
+    const startOfDay = this._calculateStartOfDay(date);
+    const calculatedTime = time
+      ? this._updateDateWithTime(startOfDay, time)
+      : startOfDay;
```
restructured the whole function, since start of day is the default value now: https://github.com/SAP/wdio-qmate-service/pull/418/commits/e59697676e7cc78cec036028b45e4a6840c94d0a#diff-6ed04468ed8e98a65f9fcda047409ea601d83a55990b3f2bd9ea889d515ee111R242

uladhsi commented:
@DIFSRIP thank you for the review

I've addressed your comments, and basically added 3 options of time anchors as argument of the `calculateWithTime`: "startOfDay" (default), "currentTime", "endOfDay". The previous time input format remains supported

DIFSRIP reviewed (approved):
Looks almost perfect now.

Just added a small suggestion.

DIFSRIP reviewed (changes requested):
pleas see last review comments
